api/openapi/engine/engine_service.yaml

```diff
@@ -0,0 +1,337 @@
+# =========================================================================================================================
+
+components:
+
+  responses:
+
+    NotOk:
+      description: Unexpected error
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ResponseInfo'
+            required: true
+
+  parameters:
+
+    enclave_identifier:
+      name: enclave_identifier
+      in: path
+      required: true
+      description: UUID, shortened UUID, or name of the enclave
+      schema:
+        type: string
+    
+    remove_all:
+      name: remove_all
+      in: query
+      required: false
+      description: If true, remove all enclaves. Default is false
+      schema:
+        type: boolean
+
+  schemas:
+
+    ResponseType:
+      type: string
+      enum:
+        - ERROR
+        - INFO
+        - WARNING 
+
+    ResponseInfo:
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/ResponseType'
+        message:
+          type: string
+        code:
+          type: integer
+          format: uint32
+      required:
+        - type
+        - message
+        - code
+
+    EngineInfo:
+      type: object
+      properties:
+        engine_version:
+          type: string
+      required:
+        - engine_version
+
+    CreateEnclave:
+      type: object
+      properties:
+        enclave_name:
+          type: string
+        api_container_version_tag:
+          type: string
+        api_container_log_level:
+          type: string
+        mode:
+          $ref: '#/components/schemas/EnclaveMode'
+    
+    EnclaveMode:
+      type: string
+      enum:
+        - TEST
+        - PRODUCTION
+    
+    EnclaveContainersStatus:
+      type: string
+      enum:
+        - RUNNING
+        - STOPPED
+        - EMPTY
+    
+    EnclaveTargetStatus:
+      type: string
+      enum:
+        - STOP
+    
+    ApiContainerStatus:
+      type: string
+      enum:
+        - RUNNING
+        - STOPPED
+        - NON_EXISTENT
+
+    EnclaveInfo:
+      type: object
+      properties:
+        enclave_uuid:
+          type: string
+        name:
+          type: string
+        shortened_uuid:
+          type: string
+        containers_status:
+          $ref: '#/components/schemas/EnclaveContainersStatus'
+        api_container_status:
+          $ref: '#/components/schemas/ApiContainerStatus'
+        api_container_info:
+          $ref: '#/components/schemas/EnclaveAPIContainerInfo'
+        api_container_host_machine_info:
+          $ref: '#/components/schemas/EnclaveAPIContainerHostMachineInfo'
+        creation_time:
+          $ref: '#/components/schemas/Timestamp'
+        mode:
+          $ref: '#/components/schemas/EnclaveMode'
+      required:
+        - enclave_uuid
+        - name
+        - shortened_uuid
+        - containers_status
+        - api_container_status
+        - creation_time
+        - mode
+
+    EnclaveAPIContainerInfo:
```

Comment:
In parallel I've added a couple of fields to `EnclaveAPIContainerInfo` and `EnclaveAPIContainerHostMachineInfo` to expose the tunnel server's ports. 
Reference is here: https://github.com/kurtosis-tech/kurtosis/pull/1849/files#diff-a24e12c03e41a9365ffc8601baddeee397de4b5547611039da5e15ac24d82a22R106 

Not sure on best timing, but we could add these fields right away, make them required and populate them with zeros to begin with?  Specifically this would be fields:

* `EnclaveAPIContainerInfo.tunnel_port_inside_enclave (integer, required)`
* `EnclaveAPIContainerHostMachineInfo.tunnel_port_on_host_machine (integer, required)`

which should avoid transitions at a later date

Reply:
Thanks for the heads up on these new fields but since they are additives we can also add them later without any break changes.